spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/TestSparkDistributionAndOrderingUtil.java

```diff
@@ -299,14 +300,70 @@ public void testRangeWritePartitionedSortedTable() {
     checkWriteDistributionAndOrdering(table, expectedDistribution, expectedOrdering);
   }
 
+  // =============================================================
+  // Distribution and ordering for copy-on-write DELETE operations
+  // =============================================================
+  //
+  // UNPARTITIONED UNORDERED
+  // -------------------------------------------------------------------------
+  // delete mode is NOT SET -> CLUSTER BY _file + LOCALLY ORDER BY _file, _pos
+  // delete mode is NONE -> unspecified distribution + empty ordering
```

Comment:
I see why you'd not introduce extra work here, since there is no table setting and the delete mode is explicit to do very little. But I think we could still introduce a local order by _file and _pos if we wanted to. If there is no order specified, we get to choose and I'd probably opt for doing a bit more work and keeping the original row order.

Reply:
I don't mind doing that. In many cases, unfortunately, it won't give us any benefits. I think frequently a single task can end up with records from multiple files and if those original files were sorted, the produced file won't be. There is also no guarantee records from the same original file will end up in the same write task. They can be scattered across all tasks.

Reply:
That's a fair point. If we aren't clustering by _file then there is probably either a mix (if there's a shuffle) or we end up with a good ordering in the first place because we're reading directly from the incoming files.

Let's leave this as is.